File: TIENDARENEGUL/src/Tienda.java
import java.util.ArrayList;
import java.util.Scanner;
import java.util.Date;


public class Tienda{
    private final String password = "flacou";
    private ArrayList<Producto> stockProductos = new ArrayList<>();
    private ArrayList<Cliente> clientes = new ArrayList<>();
    private ArrayList<Producto> productosVendidos = new ArrayList<>();
    private Scanner scanner = new Scanner(System.in);

    public void ejecutarPrograma() {
        boolean contrasenaValida = false;

        do {
            System.out.println("INICIA SESIÓN");
            System.out.println("Ingresa la contraseña: ");
            String contrasena = scanner.nextLine();

            if (contrasena.equals(password)) {
                contrasenaValida = true;
                ejecutarMenuSistema();
            } else {
                System.out.println("\nContraseña incorrecta, intenta de nuevamente. \n");
            }
        } while (!contrasenaValida);
    }

    private void ejecutarMenuSistema() {
        int opcion;

        do {
            System.out.println("\n** BIENVENIDO AL SISTEMA DE ABARROTES JOHN CENA **");
            System.out.println("1. Registrar Producto");
            System.out.println("2. Registrar Cliente");
            System.out.println("3. Mostrar Productos");
            System.out.println("4. Mostrar Clientes");
            System.out.println("5. Eliminar Producto");
            System.out.println("6. Eliminar Cliente");
            System.out.println("7. Realizar compra");
            System.out.println("8. Ver productos vendidos");
            System.out.println("9. Salir");
            System.out.println("Selecciona una opción:");

            opcion = scanner.nextInt();
            scanner.nextLine();

            switch (opcion) {
                case 1:
                    registrarProducto();
                    break;
                case 2:
                    registrarCliente();
                    break;
                case 3:
                    mostrarProductos();
                    break;
                case 4:
                    mostrarClientes();
                    break;
                case 5:
                    eliminarProducto();
                    break;
                case 6:
                    eliminarCliente();
                    break;
                case 7:
                    if (stockProductos.isEmpty()) {
                        System.out.println("No hay productos registrados. Por favor, registre productos primero.");
                        break;
                    }
                    else {
                    realizarCompra();

                    }
                    break;
                case 8:
                    mostrarProductosVendidos();
                    break;
                case 9:
                    System.out.println("Saliendo del sistema...");
                    break;
                default:
                    System.out.println("Opción no válida.");
                    break;
            }
        } while (opcion != 9);
    }

    private void registrarProducto(){
        System.out.println("Ingrese el tipo de producto a registrar:");
        System.out.println("1. Electrodomestico");
        System.out.println("2. Limpieza");
        System.out.println("3. Belleza");
        System.out.println("4. Alimento");
        System.out.println("5. Regresar al menú principal");
        int opcionTipo = scanner.nextInt();
        scanner.nextLine();

        System.out.println("Ingrese el nombre del producto:");
        String nombre = scanner.nextLine();
        System.out.println("Ingrese el precio del producto:");
        double precio = scanner.nextDouble();
        scanner.nextLine(); // Consumir la nueva línea después de nextDouble()

        System.out.println("Ingrese la fecha de importación (MM/DD/YYYY):");
        String fechaStr = scanner.nextLine();
        String fechaImportacion = new String(fechaStr);

        System.out.println("Ingrese el número de serie:");
        String numeroSerie = scanner.nextLine();

        System.out.println("Ingrese el stock del producto:");
        int stock = scanner.nextInt();
        scanner.nextLine();

        switch (opcionTipo) {
            case 1:
                System.out.println("Ingrese la marca del electrodomestico:");
                String marca = scanner.nextLine();
                System.out.println("Ingrese si tiente garantia el electrodomestico:");
                String tieneGarantia = scanner.nextLine();
                Electrodomestico electrodomestico = new Electrodomestico(nombre, precio, fechaImportacion, numeroSerie, marca, tieneGarantia, stock);
                stockProductos.add(electrodomestico);
                System.out.println("Producto electrodomestico registrado.");
                break;
            case 2:
                System.out.println("Ingrese el tipo de limpieza:");
                String tipo = scanner.nextLine();
                Limpieza limpieza = new Limpieza(nombre, precio, fechaImportacion, numeroSerie, tipo, stock);
                stockProductos.add(limpieza);
                System.out.println("Producto de limpieza registrado.");
                break;
            case 3:
                System.out.println("Ingrese el tipo de producto de belleza (organico o no organico:");
                String esOrganico = scanner.nextLine();
                Belleza belleza = new Belleza(nombre, precio, fechaImportacion, numeroSerie, esOrganico, stock);
                stockProductos.add(belleza);
                System.out.println("Producto de belleza registrado.");
                break;
            case 4:
                System.out.println("Ingrese la fecha de caducidad (MM/DD/YYYY):");
                String fechaCaducidad = scanner.nextLine();
                System.out.println("Ingrese las calorias del alimento:");
                int calorias = scanner.nextInt();
                Alimento alimento = new Alimento(nombre, precio, fechaImportacion, numeroSerie, fechaCaducidad, calorias, stock);
                stockProductos.add(alimento);
                System.out.println("Alimento registrado con éxito.");
                break;

            case 5:
                System.out.println("Regresando al menú principal...");
                ejecutarMenuSistema();
                break;
            default:
                System.out.println("Opción no válida");
                break;
        }
    }

    private void registrarCliente() {
        System.out.println("Ingrese el nombre del cliente:");
        String nombre = scanner.nextLine();
        System.out.println("Ingrese el teléfono del cliente:");
        String telefono = scanner.nextLine();

        Cliente cliente = new Cliente(nombre, telefono);
        clientes.add(cliente);
        System.out.println("Cliente registrado.");
    }

    private void mostrarProductos() {
        if (stockProductos.isEmpty()) {
            System.out.println("No hay productos en stock.");
        } else {
            System.out.println("\nProductos en stock:");
            for (Producto producto : stockProductos) {
                System.out.println(producto.getNombre() + " - Precio: " + producto.getPrecio() + " - Stock: " + producto.getStock());
            }
        }
    }

    private void mostrarClientes() {
        if (clientes.isEmpty()) {
            System.out.println("No hay clientes registrados.");
        } else {
            System.out.println("\nClientes registrados:");
            for (Cliente cliente : clientes) {
                System.out.println("Nombre: " + cliente.getNombre() + " - Dirección: " + " - Teléfono: " + cliente.getTelefono());
            }
        }
    }

    private void eliminarProducto() {
        System.out.println("Ingrese el nombre del producto a eliminar:");
        String nombre = scanner.nextLine();

        boolean encontrado = false;
        for (int i = 0; i < stockProductos.size(); i++) {
            if (stockProductos.get(i).getNombre().equalsIgnoreCase(nombre)) {
                stockProductos.remove(i);
                encontrado = true;
                System.out.println("Producto eliminado con éxito.");
                break;
            }
        }

        if (!encontrado) {
            System.out.println("Producto no encontrado en stock.");
        }
    }

    private void eliminarCliente() {
        System.out.println("Ingrese el nombre del cliente a eliminar:");
        String nombre = scanner.nextLine();

        boolean encontrado = false;
        for (int i = 0; i < clientes.size(); i++) {
            if (clientes.get(i).getNombre().equalsIgnoreCase(nombre)) {
                clientes.remove(i);
                encontrado = true;
                System.out.println("Cliente eliminado con éxito.");
                break;
            }
        }

        if (!encontrado) {
            System.out.println("Cliente no encontrado.");
        }
    }

    private void realizarCompra() {
        System.out.println("Ingrese el nombre del producto que desea comprar:");
        String nombreProducto = scanner.nextLine();

        System.out.println("Ingrese el nombre del cliente que realiza la compra:");
        String nombreCliente = scanner.nextLine();

        boolean productoEncontrado = false;
        for (Producto producto : stockProductos) {
            if (producto.getNombre().equalsIgnoreCase(nombreProducto) && producto.getStock() > 0) {
                productoEncontrado = true;
                double precioProducto = producto.getPrecio();
                producto.eliminarStock(1);
                productosVendidos.add(producto);

                System.out.println("Producto: " + nombreProducto + " - Precio: " + precioProducto);
                System.out.println("Precio total de la compra: " + precioProducto);
                System.out.println("Producto '" + nombreProducto + "' comprado por '" + nombreCliente + "'.");
                break;
            }
        }

        if (!productoEncontrado) {
            System.out.println("Producto no encontrado o sin stock disponible.");
        }
    }

    private void mostrarProductosVendidos() {
        if (productosVendidos.isEmpty()) {
            System.out.println("No hay productos vendidos.");
        } else {
            System.out.println("\nProductos vendidos:");
            for (Producto producto : productosVendidos) {
                System.out.println(producto.getNombre() + " - Precio: " + producto.getPrecio());
            }
        }
    }
}

File: TIENDARENEGULPY/Tienda.py
from TIENDARENEGUL.Alimento import Alimento
from TIENDARENEGUL.Belleza import Belleza
from TIENDARENEGUL.Cliente import Cliente
from TIENDARENEGUL.Electrodomestico import Electrodomestico
from TIENDARENEGUL.Limpieza import Limpieza


class Tienda:
    def __init__(self):
        self.password = "flacou"
        self.stock_productos = []
        self.clientes = []
        self.productos_vendidos = []
       # self.scanner = scanner()

    def ejecutar_programa(self):
        contrasena_valida = False

        while not contrasena_valida:
            print("INICIA SESIÓN")
            contrasena = input("Ingresa la contraseña: ")

            if contrasena == self.password:
                contrasena_valida = True
                self.ejecutar_menu_sistema()
            else:
                print("\nContraseña incorrecta, intenta de nuevo.\n")

    def ejecutar_menu_sistema(self):
        opcion = 0

        while opcion != 9:
            print("\n** BIENVENIDO AL SISTEMA DE ABARROTES JOHN CENA **")
            print("1. Registrar Producto")
            print("2. Registrar Cliente")
            print("3. Mostrar Productos")
            print("4. Mostrar Clientes")
            print("5. Eliminar Producto")
            print("6. Eliminar Cliente")
            print("7. Realizar compra")
            print("8. Ver productos vendidos")
            print("9. Salir")
            opcion = int(input("Selecciona una opción: "))

            if opcion == 1:
                self.registrar_producto()
            elif opcion == 2:
                self.registrar_cliente()
            elif opcion == 3:
                self.mostrar_productos()
            elif opcion == 4:
                self.mostrar_clientes()
            elif opcion == 5:
                self.eliminar_producto()
            elif opcion == 6:
                self.eliminar_cliente()
            elif opcion == 7:
                self.realizar_compra()
            elif opcion == 8:
                self.mostrar_productos_vendidos()
            elif opcion == 9:
                print("Saliendo del sistema...")
            else:
                print("Opción no válida.")

    def registrar_producto(self):
        print("Ingrese el tipo de producto a registrar:")
        print("1. Electrodomestico")
        print("2. Limpieza")
        print("3. Belleza")
        print("4. Alimento")
        print("5. Regresar al menú principal")
        opcion_tipo = int(input())

        nombre = input("Ingrese el nombre del producto: ")
        precio = float(input("Ingrese el precio del producto: "))
        fecha_importacion = input("Ingrese la fecha de importación (MM/DD/YYYY): ")
        numero_serie = input("Ingrese el número de serie: ")
        stock = int(input("Ingrese el stock del producto: "))

        if opcion_tipo == 1:
            marca = input("Ingrese la marca del electrodomestico: ")
            tiene_garantia = input("Ingrese si tiene garantia el electrodomestico: ")
            electrodomestico = Electrodomestico(nombre, precio, fecha_importacion, numero_serie, marca, tiene_garantia, stock)
            self.stock_productos.append(electrodomestico)
            print("Producto electrodomestico registrado.")
        elif opcion_tipo == 2:
            tipo = input("Ingrese el tipo de limpieza: ")
            limpieza = Limpieza(nombre, precio, fecha_importacion, numero_serie, tipo, stock)
            self.stock_productos.append(limpieza)
            print("Producto de limpieza registrado.")
        elif opcion_tipo == 3:
            es_organico = input("Ingrese el tipo de producto de belleza (organico o no organico): ")
            belleza = Belleza(nombre, precio, fecha_importacion, numero_serie, es_organico, stock)
            self.stock_productos.append(belleza)
            print("Producto de belleza registrado.")
        elif opcion_tipo == 4:
            fecha_caducidad = input("Ingrese la fecha de caducidad (MM/DD/YYYY): ")
            calorias = int(input("Ingrese las calorias del alimento: "))
            alimento = Alimento(nombre, precio, fecha_importacion, numero_serie, fecha_caducidad, calorias, stock)
            self.stock_productos.append(alimento)
            print("Alimento registrado con éxito.")
        elif opcion_tipo == 5:
            print("Regresando al menú principal...")
            self.ejecutar_menu_sistema()
        else:
            print("Opción no válida")

    def registrar_cliente(self):
        nombre = input("Ingrese el nombre del cliente: ")
        telefono = input("Ingrese el teléfono del cliente: ")
        cliente = Cliente(nombre, telefono)
        self.clientes.append(cliente)
        print("Cliente registrado.")

    def mostrar_productos(self):
        if not self.stock_productos:
            print("No hay productos en stock.")
        else:
            print("\nProductos en stock:")
            for producto in self.stock_productos:
                print(f"{producto.nombre} - Precio: {producto.precio} - Stock: {producto.stock}")

    def mostrar_clientes(self):
        if not self.clientes:
            print("No hay clientes registrados.")
        else:
            print("\nClientes registrados:")
            for cliente in self.clientes:
                print(f"Nombre: {cliente.nombre} - Teléfono: {cliente.telefono}")

    def eliminar_producto(self):
        nombre = input("Ingrese el nombre del producto a eliminar: ")
        encontrado = False
        for producto in self.stock_productos:
            if producto.nombre.lower() == nombre.lower():
                self.stock_productos.remove(producto)
                encontrado = True
                print("Producto eliminado con éxito.")
                break
        if not encontrado:
            print("Producto no encontrado en stock.")

    def eliminar_cliente(self):
        nombre = input("Ingrese el nombre del cliente a eliminar: ")
        encontrado = False
        for cliente in self.clientes:
            if cliente.nombre.lower() == nombre.lower():
                self.clientes.remove(cliente)
                encontrado = True
                print("Cliente eliminado con éxito.")
                break
        if not encontrado:
            print("Cliente no encontrado.")

    def realizar_compra(self):
        if not self.stock_productos:
            print("No hay productos registrados. Por favor, registre productos primero.")
            return

        nombre_producto = input("Ingrese el nombre del producto que desea comprar: ")
        nombre_cliente = input("Ingrese el nombre del cliente que realiza la compra: ")

        producto_encontrado = False
        for producto in self.stock_productos:
            if producto.nombre.lower() == nombre_producto.lower() and producto.stock > 0:
                producto_encontrado = True
                precio_producto = producto.precio
                producto.eliminar_stock(1)
                self.productos_vendidos.append(producto)

                print(f"Producto: {nombre_producto} - Precio: {precio_producto}")
                print(f"Precio total de la compra: {precio_producto}")
                print(f"Producto '{nombre_producto}' comprado por '{nombre_cliente}'.")
                break

        if not producto_encontrado:
            print("Producto no encontrado o sin stock disponible.")

    def mostrar_productos_vendidos(self):
        if not self.productos_vendidos:
            print("No hay productos vendidos.")
        else:
            print("\nProductos vendidos:")
            for producto in self.productos_vendidos:
                print(f"{producto.nombre} - Precio: {producto.precio}")